packages/block-editor/src/hooks/block-style-variation.js

```diff
@@ -157,6 +223,7 @@ function useBlockProps( { name, className, clientId } ) {
 		id: `variation-${ clientId }`,
 		css: variationStyles,
 		__unstableType: 'variation',
+		variation,
```

Comment:
This is new. Required to identify the variation that the override targets.

Otherwise, we could do it the long way, that is, parsing the styles as the hook does. I think.

Reply:
The tiny cost of adding it here certainly looks better than having to re-parse class names.